source/extensions/quic_listeners/quiche/platform/quic_port_utils_impl.cc

```diff
@@ -0,0 +1,53 @@
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include "extensions/quic_listeners/quiche/platform/quic_port_utils_impl.h"
+
+#include "envoy/network/address.h"
+
+#include "common/common/assert.h"
+#include "common/network/utility.h"
+
+#include "test/test_common/environment.h"
+#include "test/test_common/network_utility.h"
+
+namespace quic {
+
+int QuicPickUnusedPortOrDieImpl() {
+  // Only try to get a port for limited times.
+  std::vector<Envoy::Network::Address::IpVersion> supported_versions =
+      Envoy::TestEnvironment::getIpVersionsForTest();
+  for (size_t i = 0; i < 30000; ++i) {
```

Comment:
30000 iterations looks very high:) I think O(10) is enough.

Reply:
In other platforms, we use similar large numbers.

Reply:
SGTM.